python/docker-app-with-asg-alb/dockerized_app_cdk/asg_stack.py

```diff
@@ -0,0 +1,86 @@
+from aws_cdk.aws_ec2 import SubnetType
+from aws_cdk import (
+    aws_ec2 as ec2,
+    aws_autoscaling as autoscaling,
+    aws_elasticloadbalancingv2 as elbv2,
+    core
+    )
+
+
+class ASGStack(core.Stack):
+
+    def __init__(self, scope: core.Construct, id: str, props, **kwargs) -> None:
+        super().__init__(scope, id, **kwargs)
+
+        userdata_file = open("./userdata.sh", "rb").read()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *open*. The resource is allocated by call *builtins.open*. Execution paths that do not contain closure statements were detected. To prevent this resource leak, close the object returned by *open()* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).